File: fastify_server/src/routes/healthz/db-connection-elasticsearch.mjs
/**
 * Elasticsearch healthz routes.
 */

import {
  ElasticsearchConfig,
  checkConnection,
} from '@internal/db-connection-elasticsearch';
import { AppYamlConfig } from '@internal/app-yaml-config';
import { YamlConfigFactory, createRuntimeConfigResponse } from '@internal/yaml-config-factory';

export default async function elasticsearchRoutes(fastify) {
  fastify.get('/healthz/admin/db-connection-elasticsearch/status', async (request, reply) => {
    const config = new ElasticsearchConfig();
    const result = await checkConnection(config);
    return {
      connected: result.success,
      cluster_name: result.info?.cluster_name || null,
      version: result.info?.version?.number || null,
      error: result.error || null,
    };
  });

  fastify.get('/healthz/admin/db-connection-elasticsearch/config', async (request, reply) => {
    const configInstance = AppYamlConfig.getInstance();
    const factory = new YamlConfigFactory(configInstance);
    // @ts-ignore - computeAll is typed but we are in JS
    const result = factory.computeAll('storages.elasticsearch');
    return createRuntimeConfigResponse(result);
  });
}

File: fastapi_server/app/routes/healthz/db_connection_elasticsearch.py
"""Elasticsearch healthz routes."""

from fastapi import APIRouter, Request
from db_connection_elasticsearch import (
    ElasticsearchConfig,
    check_connection,
)
from app_yaml_config import AppYamlConfig
from yaml_config_factory import YamlConfigFactory, create_runtime_config_response

router = APIRouter(prefix="/healthz/admin/db-connection-elasticsearch", tags=["Admin"])


@router.get("/status")
async def elasticsearch_status():
    """Elasticsearch connection status."""
    config = ElasticsearchConfig()
    result = await check_connection(config)
    return {
        "connected": result.get("success", False),
        "cluster_name": result.get("info", {}).get("cluster_name") if result.get("info") else None,
        "version": result.get("info", {}).get("version", {}).get("number") if result.get("info") else None,
        "error": result.get("error"),
    }


@router.get("/config")
async def elasticsearch_config(request: Request):
    """Elasticsearch configuration."""
    config_instance = AppYamlConfig.get_instance()
    factory = YamlConfigFactory(config_instance)
    result = await factory.compute_all("storages.elasticsearch", request=request)
    return create_runtime_config_response(result)
